src/context/Zustand/provider.tsx

```diff
@@ -0,0 +1,48 @@
+import { createContext as createReactContext, useContext } from 'react';
+import { StoreName } from 'stores/names';
+import useConstant from 'use-constant';
+import { StoreApi, useStore } from 'zustand';
+import invariableStores from './invariableStores';
+import { ZustandProviderProps } from './types';
+
+export const ZustandContext = createReactContext<any | null>(null);
+
+export const ZustandProvider = ({
+    children,
+    storeSlice,
+}: ZustandProviderProps) => {
+    const storeOptions = useConstant(() => {
+        if (storeSlice) {
+            const { storeName, createStore } = storeSlice;
+
+            return { [storeName]: createStore };
+        } else {
+            return invariableStores;
+        }
+    });
+
+    return (
+        <ZustandContext.Provider value={storeOptions}>
+            {children}
+        </ZustandContext.Provider>
+    );
+};
+
+// TODO (useZustand) it would be great to check that the parent exists
+//   and if so then enable the functionality relying on this. An example
+//   where this would be good is the tables as any tables currently needs
+//   the store even if they don't allow for selection
+export const useZustandStore = <S extends Object, U>(
```

Comment:
While I understand the distinction between this file and its _hooks_ counterpart (i.e., _src/context/Zustand/hooks.ts_), it feels slightly odd to not include this Zustand hook in the _hooks_ file given the name. Again, I am not suggesting any change is required. This is merely an observation.

Reply:
Yeah - I actually think the other file should be named more like "selectors.ts" as they are hooks... but not really hooks... as they are just for the selectors. But I'm not too concerned with file naming with this PR but we should get a pattern decided soon.